users_app/views.py

```diff
@@ -1,40 +1,27 @@
-from django.shortcuts import render, redirect
-from .forms import UserLoginForm, UserRegisterForm
-from django.contrib import messages
-from django.contrib.auth import authenticate, login
 
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authtoken.models import Token
+from rest_framework.authtoken.views import ObtainAuthToken
+from .serializers import LoginSerializer, RegisterSerializer
+from rest_framework import status
 
-def login_page(request):
 
-    template = "login.html"
-    form = UserLoginForm
+class LoginAPIView(ObtainAuthToken, APIView):
+    def post(self, request, *args, **kwargs):
+        serializer = LoginSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
 
-    if request.method == "POST":
-        form = UserLoginForm(request.POST)
-        if form.is_valid():
-            username = form.cleaned_data['username']
-            password = form.cleaned_data['password']
-            user = authenticate(request, username=username, password=password)
+        user = serializer.validated_data['user']
+        token, created = Token.objects.get_or_create(user=user)
 
-            if user is not None:
-                login(request, user)
-                return redirect("calculator")
+        return Response({'token': token.key})
 
-    return render(request, template, {"form": form})
 
-
-def register_page(request):
-
-    template = "register.html"
-    form = UserRegisterForm
-
-    if request.method == "POST":
-        form = UserRegisterForm(request.POST)
-        if form.is_valid():
-            form.save()
-            messages.success(request, 'Your account has been successfully created!')
-            return redirect("login")
-        else:
-            form = UserRegisterForm()
-
-    return render(request, template, {"form": form})
+class RegisterAPIView(APIView):
+    def post(self, request, *args, **kwargs):
+        serializer = RegisterSerializer(data=request.data)
```

Comment:
ładniejszą formą jest
if not serializer.is_valid():
    return X / Raise X
Normalna logika